src/env.ts

```diff
@@ -218,6 +221,13 @@ export function setEnvConfiguration(config: Configuration, handlers: FunctionInf
     if (config.enableDDTracing !== undefined && environment[ddTracingEnabledEnvVar] === undefined) {
       environment[ddTracingEnabledEnvVar] = config.enableDDTracing;
     }
+    if (config.enableASM !== undefined) {
+      if (config.enableASM && (!config.addExtension || !config.enableDDTracing)) {
+        throw new Error("`enableASM` requires `addExtension` and `enableDDTracing` to be enabled");
```

Comment:
Some people will add the extension manually instead. I'd suggest we ignore that setting.